### PR TITLE
Bug 1907199: Bump deps, include machine-api-operator

### PR DESCRIPTION
**Pull request by nirarg**

### CODE REVIEW

File: pkg/clients/infracluster/client.go [hunk 1]
```diff
@@ -41,21 +44,30 @@ const (
...
 	UpdateVirtualMachine(namespace string, vm *kubevirtapiv1.VirtualMachine) (*kubevirtapiv1.VirtualMachine, error)
-	PatchVirtualMachine(namespace string, name string, pt types.PatchType, data []byte, subresources ...string) (result *kubevirtapiv1.VirtualMachine, err error)
-	RestartVirtualMachine(namespace string, name string) error
-	StartVirtualMachine(namespace string, name string) error
-	StopVirtualMachine(namespace string, name string) error
 	CreateSecret(namespace string, newSecret *corev1.Secret) (*corev1.Secret, error)
 }
 
+var (
+	vmRes = schema.GroupVersionResource{
```
Comment (bardielle) [inline pkg/clients/infracluster/client.go]: vmResource

File: pkg/clients/infracluster/client.go [hunk 2]
```diff
@@ -41,21 +44,30 @@ const (
...
+		Group:    kubevirtapiv1.GroupVersion.Group,
+		Version:  kubevirtapiv1.GroupVersion.Version,
+		Resource: "virtualmachines",
+	}
+	vmiRes = schema.GroupVersionResource{
```
Comment (bardielle) [inline pkg/clients/infracluster/client.go]: vmiResource

File: pkg/clients/infracluster/client.go [hunk 3]
```diff
@@ -89,52 +97,128 @@ func New(tenantClusterKubernetesClient tenantcluster.Client) (Client, error) {
...
 
-func (c *client) GetVirtualMachineInstance(namespace string, name string, options *k8smetav1.GetOptions) (*kubevirtapiv1.VirtualMachineInstance, error) {
-	return c.kubevirtClient.VirtualMachineInstance(namespace).Get(name, options)
+func (c *client) ListVirtualMachine(namespace string, options metav1.ListOptions) (*kubevirtapiv1.VirtualMachineList, error) {
+	resp, err := c.listResource(namespace, vmRes, options)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list VirtualMachine")
+	}
+	var vmList kubevirtapiv1.VirtualMachineList
+	err = c.fromUnstructedListToInterface(*resp, &vmList, "VirtualMachineList")
```
Comment (bardielle) [inline pkg/clients/infracluster/client.go:142]: If there isn't any VM the vmList would be return an empty list? or None? if None need to initialize it to an empty list on line 139

File: Dockerfile.rhel [hunk 4]
```diff
@@ -13,15 +13,20 @@
...
+	version="$version" \
+	release="$release" \
         architecture="x86_64" \
         summary="cluster-api-provider-kubevirt" \
         maintainer="ocpcnvteam <ocpcnvteam@redhat.com>"
         
 WORKDIR /go/src/github.com/kubevirt/cluster-api-provider-kubevirt
 COPY . .
-RUN make build NO_DOCKER=1
+RUN NO_DOCKER=1 make build
```
Comment (bardielle) [inline Dockerfile.rhel]: why does it matter?

File: cmd/manager/main.go [hunk 5]
```diff
@@ -136,7 +128,7 @@ func main() {
 	}
 
 	// Initialize infra-cluster clients
-	infraClusterClient, err := infracluster.New(tenantClusterClient)
+	infraClusterClient, err := infracluster.New(context.Background(), tenantClusterClient)
```
Comment (bardielle) [inline cmd/manager/main.go:131]: Dont think it's related to that commit

File: cmd/manager/main.go [hunk 6]
```diff
@@ -42,23 +43,19 @@ var (
 )
 
 func main() {
-	var printVersion bool
-	flag.BoolVar(&printVersion, "version", false, "print version and exit")
-
-	// TODO Add relevant flags as written in klog initFlags
-	//klog.InitFlags(nil)
+	klog.InitFlags(nil)
```
Comment (bardielle) [inline cmd/manager/main.go:46]: I think that this is the rout cause of your error - can you test it without the last commit but only with that?

File: pkg/clients/infracluster/client.go (hunk 1 above)
Comment (nirarg) [reply pkg/clients/infracluster/client.go]: Changed

File: pkg/clients/infracluster/client.go (hunk 2 above)
Comment (nirarg) [reply pkg/clients/infracluster/client.go]: Changed

File: Dockerfile.rhel (hunk 4 above)
Comment (nirarg) [reply Dockerfile.rhel]: Removed Dockerfile changes

File: cmd/manager/main.go (hunk 5 above)
Comment (nirarg) [reply cmd/manager/main.go:131]: Changed the commits

File: cmd/manager/main.go (hunk 6 above)
Comment (nirarg) [reply cmd/manager/main.go:46]: Removed Dockerfile changes and validated

File: pkg/clients/infracluster/client.go (hunk 3 above)
Comment (nirarg) [reply pkg/clients/infracluster/client.go:142]: vmList is not a pointer, so its default value is empty struct (can't be nil)

**Comment by nirarg** [comment]
@JoelSpeed this PR bump `machine-api-operator` version and removing the replacement

File: pkg/clients/infracluster/client.go (hunk 1 above)
Comment (bardielle) [reply pkg/clients/infracluster/client.go]: Can you squash you last commit with that commit?

**Review by bardielle — APPROVED** [review]
/lgtm

**Pull request merged**